src/rapids_dependency_file_generator/rapids_dependency_file_generator.py

```diff
@@ -413,43 +429,78 @@ def make_dependency_files(parsed_config, config_file_path, to_stdout):
                                 # indicate that for some matrix elements no
                                 # packages should be installed.
                                 dependencies.extend(
-                                    specific_matrices_entry["packages"] or []
+                                    specific_matrices_entry.get("packages", []) or []
                                 )
 
                         if not found:
-                            if fallback_entry:
-                                dependencies.extend(fallback_entry["packages"] or [])
+                            if fallback_entry is not None:
+                                dependencies.extend(
+                                    fallback_entry.get("packages", []) or []
+                                )
                             else:
                                 raise ValueError(
                                     f"No matching matrix found in '{include}' for: {matrix_combo}"
                                 )
 
                 # Dedupe deps and print / write to filesystem
                 full_file_name = get_filename(file_type, file_key, matrix_combo)
-                deduped_deps = dedupe(dependencies)
 
-                output_dir = (
-                    "."
-                    if to_stdout
-                    else get_output_dir(file_type, config_file_path, file_config)
-                )
-                contents = make_dependency_file(
+                output_dir = get_output_dir(file_type, config_file_path, file_config)
+
+                output_file_path = os.path.join(output_dir, full_file_name)
+
+                if results.get(output_file_path, None) is None:
+                    if file_type != str(OutputTypes.PYPROJECT):
+                        results[output_file_path] = ""
+                    else:
+                        # pyproject.toml needs to be modified in place instead of built from scratch.
+                        with open(output_file_path) as f:
+                            results[output_file_path] = tomlkit.load(f)
+                            # Append `-cuXX` to `[package.name]`
+                            results[output_file_path]["project"][
+                                "name"
+                            ] = name_with_cuda_suffix(
+                                results[output_file_path]["project"]["name"],
+                                matrix_combo.get("cuda", None),
+                                cuda_suffix,
+                            )
```

Comment:
When I first saw the function I actually thought the goal was to handle _dependency_ suffixes rather than package name suffixes. I'm even less comfortable putting this logic into this tool. Package renaming to support our wheels-specific workflows is definitely scope creep. I'd love to find a better solution to what we're currently doing in our wheel builds, but I don't think this is it.